### PR TITLE
Add remoteRead custom headers

### LINKED ISSUE #4320: Add remoteRead custom headers
**What is missing?**
The possibility to add the remoteRead custom HTTP headers introduced in Prometheus v2.26.0 (see [Changelog](https://github.com/prometheus/prometheus/releases/tag/v2.26.0) or Updated [Configuration](https://prometheus.io/docs/prometheus/2.26/configuration/configuration/#remote_read)). New Config Option:

```
# Custom HTTP headers to be sent along with each remote read request.
# Be aware that headers that are set by Prometheus itself can't be overwritten.
headers:
  [ <string>: <string> ... ]
```

**Why do we need it?**
To support the Prometheus config option.

**Anything else we need to know?**:
I will implement the change and create a PR.

### PR DESCRIPTION
**Pull request by MonolithProjects**
## Description

This PR is adding possibility to configure custom HTTP headers in Remote Read, introduced in Prometheus 2.26.0
Closes #4320 

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [x] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

- Add option to configure custom HTTP headers in Remote Read.

```release-note

```

### CODE REVIEW

**Comment by MonolithProjects** [comment]
ah thx @fpetkovski , go 1.17 really did help. @simonpasquier i rebased again and regenerated the files....

File: scripts/push-docker-image.sh [hunk 1]
```diff
@@ -77,7 +77,8 @@ for r in ${OPERATORS} ${RELOADERS}; do
 				  "${r}:${TAG}-amd64" \
 				  "${r}:${TAG}-arm64" \
 				  "${r}:${TAG}-arm" \
-				  "${r}:${TAG}-ppc64le"
+				  "${r}:${TAG}-ppc64le" \
+				  "${r}:${TAG}-s390x"
```
Comment (paulfantom) [inline scripts/push-docker-image.sh:81]: This change was already introduced in https://github.com/prometheus-operator/prometheus-operator/pull/4351.

Comment (PhilipGough) [reply scripts/push-docker-image.sh:81]: @MonolithProjects I think we will want to rebase this against master once more and squash the commits.

Comment (MonolithProjects) [reply scripts/push-docker-image.sh:81]: @PhilipGough no problem. will do it

Comment (MonolithProjects) [reply scripts/push-docker-image.sh]: @PhilipGough now it should be hopefully ok

**Review by PhilipGough — APPROVED** [review]
lgtm

**Comment by paulfantom** [comment]
lgtm, but first merge conflicts need to be resolved.

**Comment by MonolithProjects** [comment]
hi @paulfantom , i rebased and regenerated the files again.

File: jsonnet/prometheus-operator/prometheuses-crd.json [hunk 2]
```diff
@@ -1,3 +1,6 @@
+<<<<<<< HEAD
```
Comment (paulfantom) [inline jsonnet/prometheus-operator/prometheuses-crd.json:1]: Seems like rebase went wrong here.

Comment (MonolithProjects) [reply jsonnet/prometheus-operator/prometheuses-crd.json]: @paulfantom ah sorry for that. Now it should be OK

**Comment by paulfantom** [comment]
Thank you! :+1:

**Pull request merged**